src/app/components/core/control-dispatcher/list-control-dispatcher/control-tasks-list/control-tasks-list.component.ts

```diff
@@ -0,0 +1,200 @@
+            const categoryName = script?.configuration?.categoryName;
+            // close modal
+            this.closeModal('confirmation-dialog');
+            // delete category
+            this.configurationService.deleteCategory(categoryName).subscribe((data: any) => {
+              this.alertService.success(data.message);
+            }, error => {
+              /** request completed */
+              this.ngProgress.done();
+              // close modal
+              this.closeModal('confirmation-dialog');
+              if (error.status === 0) {
+                console.log('service down ', error);
+              } else {
+                this.alertService.error(error.statusText);
+              }
+            });
+          }, error => {
+            /** request completed */
+            this.ngProgress.done();
+            // close modal
+            this.closeModal('confirmation-dialog');
+            if (error.status === 0) {
+              console.log('service down ', error);
+            } else {
+              this.alertService.error(error.statusText);
+            }
+          }, () => {
+            this.getControlScripts();
+            this.cd.detectChanges();
+          });
+      }, error => {
+        /** request completed */
+        this.ngProgress.done();
+        // close modal
+        this.closeModal('confirmation-dialog');
+        if (error.status === 0) {
+          console.log('service down ', error);
+        } else {
+          this.alertService.error(error.statusText);
+        }
+      });
+  }
+
+  showParameters(val) {
+    let config = JSON.parse(val);
+    const params =  config.map(c => c.values);
+    const parameters = [];
+    params.forEach(element => {
+      for (const [key, value] of Object.entries(element)) {
+        parameters.push({key, value})
+      }
+    });
+    return parameters;
+  }
+
+  showService(val) {
+    let config = JSON.parse(val);
+    return [...new Set(config.map(c => c.service))];
+  }
+
+  updateScheduleStatus(schedule) {
+    if (!schedule.enabled) {
+      this.enableSchedule(schedule);
+    } else {
+      this.disableSchedule(schedule);
+    }
+  }
+
+  enableSchedule(schedule) {
+    /** request started */
+    this.ngProgress.start();
+    this.schedulesService.enableScheduleByName(schedule.name).
+      subscribe(
+        (data: any) => {
+          /** request completed */
+          this.ngProgress.done();
+          this.alertService.success(data.message, true);
+          // update local reference of schedule
+          schedule.enabled = true;
+          this.cd.detectChanges();
+        },
+        error => {
+          /** request completed */
+          this.ngProgress.done();
+          if (error.status === 0) {
+            console.log('service down ', error);
+          } else {
+            this.alertService.error(error.statusText);
+          }
+        });
+  }
+
+  disableSchedule(schedule) {
```

Comment:
we can have 2 if else instead of having 2 separate copy of enable/disable

Reply:
enable/disable are 2 different api calls so 2separate methods 1 for enable schedule and1 for disable schedule